api/api.go

```diff
@@ -114,3 +126,40 @@ func NewAPI(config *conf.Configuration, db *gorm.DB, mailer *mailer.Mailer) *API
 	api.handler = corsHandler.Handler(mux)
 	return api
 }
+
+func (a *API) trace(f func(context.Context, http.ResponseWriter, *http.Request)) func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+		id := uuid.NewRandom().String()
+		log := a.log.WithField("request_id", id)
+
+		log = log.WithFields(logrus.Fields{
+			"method": r.Method,
+			"path":   r.URL.Path,
+		})
+
+		// optionally add the user id stuff
+		claims := getClaims(ctx)
+		if claims != nil {
+			log = log.WithFields(logrus.Fields{
```

Comment:
I'm guessing this accumulates entries and it won't lose method and path, doesn't it?